src/EditorFeatures/CSharpTest/CodeActions/ConvertLocalFunctionToMethod/ConvertLocalFunctionToMethodTests.cs

```diff
@@ -0,0 +1,439 @@
+        public async Task TestCaptures()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    static void Use<T>(T a) {}
+    static void Use<T>(ref T a) {}
+
+    static void LocalFunction() {} // trigger rename
+
+    void M<T1, T2>(T1 param1, T2 param2)
+        where T1 : struct
+        where T2 : struct
+    {
+        var local1 = 0;
+        var local2 = 0;
+        void [||]LocalFunction()
+        {
+            Use(param1);
+            Use(ref param2);
+            Use(local1);
+            Use(ref local2);
+            Use(this);
+            LocalFunction();
+        }
+        LocalFunction();
+        System.Action x = LocalFunction;
+    }
+}",
+@"class C
+{
+    static void Use<T>(T a) {}
+    static void Use<T>(ref T a) {}
+
+    static void LocalFunction() {} // trigger rename
+
+    void M<T1, T2>(T1 param1, T2 param2)
+        where T1 : struct
+        where T2 : struct
+    {
+        var local1 = 0;
+        var local2 = 0;
+        LocalFunction1(param1, ref param2, local1, ref local2);
+        System.Action x = () => LocalFunction1(param1, ref param2, local1, ref local2);
+    }
+
+    private void LocalFunction1<T1, T2>(T1 param1, ref T2 param2, int local1, ref int local2)
+        where T1 : struct
+        where T2 : struct
+    {
+        Use(param1);
+        Use(ref param2);
+        Use(local1);
+        Use(ref local2);
+        Use(this);
+        LocalFunction1(param1, ref param2, local1, ref local2);
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsConvertLocalFunctionToMethod)]
+        public async Task TestTypeParameters1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C<T0>
+{
+    static void LocalFunction() {} // trigger rename
+
+    void M<T1, T2>(int i)
+        where T1 : struct
+    {
+        void Local1<T3, T4>()
+            where T4 : struct
+        {
+            void [||]LocalFunction<T5, T6>(T5 a, T6 b)
+                where T5 : struct
+            {
+                _ = typeof(T2);
+                _ = typeof(T4);
+                LocalFunction(a, b);
+                System.Action<T5, T6> x = LocalFunction;
+            }
+            LocalFunction<byte, int>(5, 6);
+            LocalFunction(5, 6);
+        }
+    }
+}",
+@"class C<T0>
+{
+    static void LocalFunction() {} // trigger rename
+
+    void M<T1, T2>(int i)
+        where T1 : struct
+    {
+        void Local1<T3, T4>()
+            where T4 : struct
+        {
+            LocalFunction1<T2, T4, byte, int>(5, 6);
+            LocalFunction1<T2, T4, int, int>(5, 6);
+        }
+    }
+
+    private static void LocalFunction1<T2, T4, T5, T6>(T5 a, T6 b)
+        where T4 : struct
+        where T5 : struct
+    {
+        _ = typeof(T2);
+        _ = typeof(T4);
+        LocalFunction1<T2, T4, T5, T6>(a, b);
+        System.Action<T5, T6> x = (T5 a, T6 b) => LocalFunction1<T2, T4, T5, T6>(a, b);
```

Comment:
- Here we're introducing names so we should check uniqueness
- The simplifier doesn't seem to pick up on this case, I thought it's because of errors, but it's not. maybe I'm mistaken.